src/Core/Domain/Carrier/Command/EditCarrierCommand.php

```diff
@@ -143,4 +148,64 @@ public function setLogoPathName(?string $logoPathName): self
 
         return $this;
     }
+
+    public function isShippingHandling(): ?bool
+    {
+        return $this->isShippingHandling ?? null;
+    }
+
+    public function setIsShippingHandling(bool $isShippingHandling): self
+    {
+        $this->isShippingHandling = $isShippingHandling;
+
+        return $this;
+    }
+
+    public function isFree(): ?bool
+    {
+        return $this->isFree ?? null;
+    }
+
+    public function setIsFree(bool $isFree): self
+    {
+        $this->isFree = $isFree;
+
+        return $this;
+    }
+
+    public function getShippingMethod(): ?int
+    {
+        return $this->shippingMethod ?? null;
+    }
+
+    public function setShippingMethod(int $shippingMethod): self
+    {
+        $this->shippingMethod = $shippingMethod;
+
+        return $this;
+    }
+
+    public function getIdTaxRuleGroup(): ?int
+    {
+        return $this->idTaxRuleGroup ?? null;
+    }
+
+    public function setIdTaxRuleGroup(int $idTaxRuleGroup): self
+    {
+        $this->idTaxRuleGroup = $idTaxRuleGroup;
+
+        return $this;
+    }
+
+    public function getRangeBehavior(): ?bool
+    {
+        return $this->rangeBehavior ?? null;
+    }
+
+    public function setRangeBehavior(bool $rangeBehavior): self
```

Comment:
Woooow, this is so not adapted to the usage 😅 But it's already badly design in legacy class:

```php
    /** @var bool Behavior for out-of-range weights: true to disable carrier, false to apply the cost of the highest defined range */
    public $range_behavior;
```

I wonder if we could improve this in the CQRS layer to make it more understandable and self-explanatory. This could be renamed into `useHighestRange` for example, when set to true it uses the highest range, when set to false the carrier is disabled Or maybe `disabledWhenOutOfRange` would be clearer? When set to true you understand that it is disabled and that it's related to the out of range case

The alternative to use the highest one is less explicit and more about a supposedly known convention, but I don't think we can find a way that is clear for both. Unless we completely change the structure and stop using a boolean (in CQRS only), we can use an integer, a string or an enum with two explicit const names: `OutOfRangeBehaviour::USE_HIGHEST_RANGE` and `OutOfRangeBehaviour::DISABLED` then the handler convert them into the legacy boolean

Reply:
@jolelievre 

> Unless we completely change the structure and stop using a boolean (in CQRS only), we can use an integer, a string or an enum with two explicit const names: OutOfRangeBehaviour::USE_HIGHEST_RANGE and OutOfRangeBehaviour::DISABLED then the handler convert them into the legacy boolean

This sounds like the best option, and it also opens the world to another possibilities, like `OutOfRangeBehaviour::SCALE_PROGRESSIVELY`.